osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -641,6 +665,8 @@ public override void OnSuspending(ScreenTransitionEvent e)
             // Without this, it's possible for a transfer to happen while we are not the current screen.
             transferRulesetValue();
 
+            backgroundBlurLevel.ValueChanged -= applyBackgroundBlur;
+
```

Comment:
Don't do this. Put a condition in the value change handling method to check whether `this.IsCurrentScreen`.

Reply:
Last commits should have (hopefully) fixed all your comments